sdk/core/Azure.Core/src/Pipeline/HttpPipelineTransport.cs

```diff
@@ -1,16 +1,36 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System.Threading;
+using System.ClientModel.Primitives;
 using System.Threading.Tasks;
 
 namespace Azure.Core.Pipeline
 {
     /// <summary>
     /// Represents an HTTP pipeline transport used to send HTTP requests and receive responses.
     /// </summary>
-    public abstract class HttpPipelineTransport
+    public abstract partial class HttpPipelineTransport
     {
+        private readonly PipelineTransport _transport;
+
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        protected HttpPipelineTransport()
+        {
+            _transport = new AzureCorePipelineTransport(this);
+        }
+
+        internal void ProcessInternal(HttpMessage message)
```

Comment:
This effectively retroactively implements the template method and allows us to use ClientModel PipelineTransport validations in the Azure.Core abstract transport type.  This gives us buffering in the default pipeline even for custom subtypes of HttpPipelineTransport.